app/models/space.rb

```diff
@@ -0,0 +1,20 @@
+class Space
+  include DataMapper::Resource
+
+
+  has n, :users, through: Resource
+
+  property :id,     Serial
+  property :name,  String
+  property :location, String
+  property :description, String
+  property :price_per_night, Integer
+  property :available_from, Date
+  property :available_to, Date
+
+  validates_presence_of :name, :location, :description, :price_per_night, :available_from, :available_to
+
+  
+end
```

Comment:
BTW this class is where I'd put the `filter` method, to keep it out of your controller. After all, `filter` is essentially a Ruby proxy for a database query, and this `Space` object is hooked up to the ORM. So put it here